WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+                        Text($0.name)
+                    }
+                }
+                .pickerStyle(.segmented)
+                .padding()
+                Spacer()
+                selectedPackageTypeView
+                Spacer()
+                Button(Localization.addPackage) {
+                    // add package
+                }
+                .buttonStyle(PrimaryButtonStyle())
+                .disabled(addPackageButtonDisabled)
+                .padding()
+            }
+            .toolbar {
+                ToolbarItem(placement: .topBarLeading) {
+                    Button(action: {
+                        presentationMode.wrappedValue.dismiss()
+                    }, label: {
+                        Text(Localization.cancel)
+                    })
+                }
+            }
+            .navigationTitle(Localization.addPackage)
+            .navigationBarTitleDisplayMode(.inline)
+        }
+        .navigationViewStyle(.stack)
+    }
+
+    @ViewBuilder
+    private var selectedPackageTypeView: some View {
+        ScrollView {
+            switch selectedPackageType {
+            case .custom:
+                customPackageView
+            case .carrier:
+                carrierPackageView
+            case .saved:
+                savedPackageView
+            }
+        }
+    }
+
+    enum PackageType: CaseIterable {
+        case box, envelope
+        var name: String {
+            switch self {
+            case .box:
+                return Localization.box
+            case .envelope:
+                return Localization.envelope
+            }
+        }
+    }
+
+    @State var packageType: PackageType = .box
+    @State var showSaveTemplate: Bool = false
+    @State var length: String = ""
+    @State var fieldValues: [WooShippingAddPackageDimensionView.DimensionType: String] = [:]
+
+    private var customPackageView: some View {
+        VStack(alignment: .leading) {
+            HStack {
+                Text(Localization.packageType)
+                    .font(.subheadline)
+                Spacer()
+            }
+            // type selection
+            Menu {
+                // show selection
+                ForEach(PackageType.allCases, id: \.self) { option in
+                    Button {
+                        packageType = option
+                    } label: {
+                        Text(option.name)
+                            .font(.body)
+                    }
+                }
+            } label: {
+                HStack {
+                    // text
+                    Text(packageType.name)
+                    // arrows
+                    Spacer()
+                    Image(systemName: "chevron.up.chevron.down")
+                }
+                .padding()
+            }
+            .roundedBorder(cornerRadius: 8, lineColor: Color(.separator), lineWidth: 1)
+            HStack(spacing: 8) {
```

Comment:
This is a clever way to lay out the three dimension fields! I found the binding a little tricky to follow at first but I can see how it keeps things concise.

nit: What do you think of using an `AdaptiveStack` instead of `HStack` here, so it switches to a `VStack` with larger dynamic text sizes? That way each field would have more space.